bigquery_etl/shredder/bug1751979.py

```diff
@@ -370,3 +370,134 @@
+  sanitize_scalar(search_content_unknown_sum) AS search_content_unknown_sum
+)
+"""
+
+BUG_1751979_SEARCH_CLIENTS_DAILY_V8_UDFS = r"""
+CREATE TEMP FUNCTION sanitize_engine_source(
+  engine STRING, `source` STRING
+) AS ((
+    WITH base AS (
+      SELECT CONCAT(engine, '.', `source`) AS key
+    ),
+    parsed AS (
+      SELECT
+        *,
+        REGEXP_EXTRACT(key, "([^.]+[.]in-content[:.][^:]+:).*") AS prefix,
+        REGEXP_EXTRACT(key, "[^.]+[.]in-content[:.][^:]+:(.*)") AS code,
+      FROM
+        base
+    ),
+    scrubbed AS (
+      SELECT
+          IF(
+            prefix IS NULL
+            OR code IN (
+              "none",
+              "other",
+              "hz",
+              "h_",
+              "MOZ2",
+              "MOZ4",
+              "MOZ5",
+              "MOZA",
+              "MOZB",
+              "MOZD",
+              "MOZE",
+              "MOZI",
+              "MOZM",
+              "MOZO",
+              "MOZT",
+              "MOZW",
+              "MOZSL01",
+              "MOZSL02",
+              "MOZSL03",
+              "firefox-a",
+              "firefox-b",
+              "firefox-b-1",
+              "firefox-b-ab",
+              "firefox-b-1-ab",
+              "firefox-b-d",
+              "firefox-b-1-d",
+              "firefox-b-e",
+              "firefox-b-1-e",
+              "firefox-b-m",
+              "firefox-b-1-m",
+              "firefox-b-o",
+              "firefox-b-1-o",
+              "firefox-b-lm",
+              "firefox-b-1-lm",
+              "firefox-b-lg",
+              "firefox-b-huawei-h1611",
+              "firefox-b-is-oem1",
+              "firefox-b-oem1",
+              "firefox-b-oem2",
+              "firefox-b-tinno",
+              "firefox-b-pn-wt",
+              "firefox-b-pn-wt-us",
+              "ubuntu",
+              "ffab",
+              "ffcm",
+              "ffhp",
+              "ffip",
+              "ffit",
+              "ffnt",
+              "ffocus",
+              "ffos",
+              "ffsb",
+              "fpas",
+              "fpsa",
+              "ftas",
+              "ftsa",
+              "newext",
+              "monline_dg",
+              "monline_3_dg",
+              "monline_4_dg",
+              "monline_7_dg"
+            ),
+            key,
+            CONCAT(
+              prefix,
+              "other.scrubbed.",
+              -- Add a hash of code to avoid multiple rows ending up with the same
+              -- (client_id, sanitized_engine, sanitized_source) value, violating
+              -- the table's contract.
+              SUBSTR(TO_HEX(SHA256(code)), 1, 4))
+          ) AS key
+      FROM
+        parsed
+    )
+    SELECT AS STRUCT
+      SUBSTR(_key, 0, pos - 2) AS engine,
+      SUBSTR(_key, pos) AS `source`,
+    FROM
+      scrubbed,
+      UNNEST([REPLACE(key, 'in-content.', 'in-content:')]) AS _key,
+      UNNEST([LENGTH(REGEXP_EXTRACT(_key, '.+?[.].'))]) AS pos
```

Comment:
seems like the `.+` vs `.+?` change could still result in duplicate client/engine/source entries, but we know it doesn't in practice, that's good enough.

Reply:
I think likelihood is low, and this is a stopgap for now. I think we should fully rerun this in a month, after we're sure `clients_daily` is scrubbed through all history.